lib/vm.js

```diff
@@ -97,6 +128,31 @@ function validateObject(prop, propName) {
     throw new ERR_INVALID_ARG_TYPE(propName, 'Object', prop);
 }
 
+function getRunInContextArgs(options = {}) {
+  if (typeof options !== 'object' || options === null) {
+    throw new ERR_INVALID_ARG_TYPE('options', 'Object', options);
+  }
+  const {
+    displayErrors = true,
+    breakOnSigint = false
+  } = options;
+
+  let timeout = options.timeout;
+  if (timeout === undefined) {
+    timeout = -1;
+  } else {
+    timeout = Number(timeout);
+    if (timeout <= 0) {
+      throw new ERR_OUT_OF_RANGE('options.timeout', 'a positive number');
```

Comment:
Can you pass the timeout to the constructor so it gets displayed in the error message?